### PR TITLE
Fix some configure error in Windows native build

### PR DESCRIPTION
**Pull request by zouboan**
## Summary
1. there's a 'input line is too long' error when call CLEAN ;
2. there's no uname support in MinGW's  lib ;
3. Make.defs and .config deleted too early In distclean, and Makefile.host of pic32 and cxd56 did'nt considering Windows native build.
4. when pipe input to sed, there's a bug with a pipe being in the string, so in Windows native build we can't use `TOPDIR := ${shell echo $(CURDIR) | sed -e 's/ /\\ /g'}`
[https://stackoverflow.com/questions/804646/how-do-you-strip-quotes-out-of-an-echoed-string-in-a-windows-batch-file](url)
## Impact
Windows native build
## Testing
MinGW + GnuWin32 + custom_apps + Windows native build configs

### CODE REVIEW

File: tools/Config.mk [hunk 1]
```diff
@@ -523,12 +523,18 @@ ifeq ($(CONFIG_ARCH_COVERAGE),y)
 endif
 
 ifeq ($(CONFIG_WINDOWS_NATIVE),y)
+
+define NEWLINE
```
Comment (xiaoxiang781216) [inline tools/Config.mk:527]: why need append NEWLINE to the command?

File: tools/Config.mk [hunk 2]
```diff
@@ -523,12 +523,18 @@ ifeq ($(CONFIG_ARCH_COVERAGE),y)
...
+
+endef
+
 define CLEAN
 	$(Q) if exist *$(OBJEXT) (del /f /q *$(OBJEXT))
 	$(Q) if exist *$(LIBEXT) (del /f /q *$(LIBEXT))
 	$(Q) if exist *~ (del /f /q *~)
 	$(Q) if exist (del /f /q  .*.swp)
-	$(Q) if exist $(OBJS) (del /f /q $(OBJS))
+	$(foreach OBJ, $(OBJS), $(NEWLINE) $(call DELFILE,$(OBJ)))
```
Comment (xiaoxiang781216) [inline tools/Config.mk:537]: why not call del like before?

Comment (zouboan) [reply tools/Config.mk:537]: there's a 'input line is too long' error when use if exist $(OBJS)

File: tools/Config.mk (hunk 1 above)
Comment (zouboan) [reply tools/Config.mk:527]: to dismiss an error, for example OBJS={A,B,C} , in the runing the statement $(foreach OBJ, $(OBJS), $(NEWLINE) $(call DELFILE,$(OBJ))) will be expanded as:
if exist A del /f /q A   **if** exist B del /f /q B    if exist C del /f /q C     there's an error:
there should no 'if'   ^
and the problem can not be solved neitherr  'space' nor ';'

File: tools/Config.mk (hunk 2 above)
Comment (xiaoxiang781216) [reply tools/Config.mk:537]: I mean we can keep del:
```
$(foreach OBJ, $(OBJS), if exist $(OBJ) (del /f /q $(OBJ)))
```

Comment (zouboan) [reply tools/Config.mk:537]: without NEWLINE there will be an error: 'there should be no if'
in fact i have tested `$(foreach OBJ, $(OBJS), if exist $(OBJ) (del /f /q $(OBJ)))
` and `$(foreach OBJ, $(OBJS), $(NEWLINE) if exist $(OBJ) (del /f /q $(OBJ)))
` they all emits error in Windows native build

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by fjpanag** [comment]
It seems that this PR introduced another error in the build system:

```
make  -C nuttx/tools -f Makefile.host clean
make[1]: Entering directory '/home/fotis/C/CBox-NX/nuttx/tools'
/bin/sh: line 1: /home/fotis/C/CBox-NX/nuttx/tools/../tools/incdir: No such file or directory
/bin/sh: line 1: /home/fotis/C/CBox-NX/nuttx/tools/../tools/incdir: No such file or directory
rm -f  b16
rm -f  b16.exe
rm -f  bdf-converter
rm -f  bdf-converter.exe
rm -f  cmpconfig
rm -f  cmpconfig.exe
rm -f  cnvwindeps
rm -f  cnvwindeps.exe
rm -f  convert-comments
rm -f  convert-comments.exe
rm -f  configure
rm -f  configure.exe
rm -f  detab
rm -f  detab.exe
rm -f  gencromfs
rm -f  gencromfs.exe
rm -f  initialconfig
rm -f  initialconfig.exe
rm -f  lowhex
rm -f  lowhex.exe
rm -f  Make.dep
rm -f  mkconfig
rm -f  mkconfig.exe
rm -f  mkdeps
rm -f  mkdeps.exe
rm -f  mksymtab
rm -f  mksymtab.exe
rm -f  mksyscall
rm -f  mksyscall.exe
rm -f  mkversion
rm -f  mkversion.exe
rm -f  nxstyle
rm -f  nxstyle.exe
rm -f  rmcr
rm -f  rmcr.exe
rm -f  jlink-nuttx.so
rm -f  jlink-nuttx.dll
rm -rf *.dSYM
make -C pic32 -f Makefile.host clean
make[2]: Entering directory '/home/fotis/C/CBox-NX/nuttx/tools/pic32'
Makefile.host:21: /home/fotis/C/CBox-NX/nuttx/tools/../Make.defs: No such file or directory
make[2]: *** No rule to make target '/home/fotis/C/CBox-NX/nuttx/tools/../Make.defs'.  Stop.
make[2]: Leaving directory '/home/fotis/C/CBox-NX/nuttx/tools/pic32'
make[1]: *** [Makefile.host:276: clean] Error 2
```

**Comment by zouboan** [comment]
> It seems that this PR introduced another error in the build system:
> 
> ```
> make  -C nuttx/tools -f Makefile.host clean
> make[1]: Entering directory '/home/workspace/nuttx/tools'
> /bin/sh: line 1: /home/workspace/nuttx/tools/../tools/incdir: No such file or directory
> /bin/sh: line 1: /home/workspace/nuttx/tools/../tools/incdir: No such file or directory
> rm -f  b16
> rm -f  b16.exe
> rm -f  bdf-converter
> rm -f  bdf-converter.exe
> rm -f  cmpconfig
> rm -f  cmpconfig.exe
> rm -f  cnvwindeps
> rm -f  cnvwindeps.exe
> rm -f  convert-comments
> rm -f  convert-comments.exe
> rm -f  configure
> rm -f  configure.exe
> rm -f  detab
> rm -f  detab.exe
> rm -f  gencromfs
> rm -f  gencromfs.exe
> rm -f  initialconfig
> rm -f  initialconfig.exe
> rm -f  lowhex
> rm -f  lowhex.exe
> rm -f  Make.dep
> rm -f  mkconfig
> rm -f  mkconfig.exe
> rm -f  mkdeps
> rm -f  mkdeps.exe
> rm -f  mksymtab
> rm -f  mksymtab.exe
> rm -f  mksyscall
> rm -f  mksyscall.exe
> rm -f  mkversion
> rm -f  mkversion.exe
> rm -f  nxstyle
> rm -f  nxstyle.exe
> rm -f  rmcr
> rm -f  rmcr.exe
> rm -f  jlink-nuttx.so
> rm -f  jlink-nuttx.dll
> rm -rf *.dSYM
> make -C pic32 -f Makefile.host clean
> make[2]: Entering directory '/home/workspace/nuttx/tools/pic32'
> Makefile.host:21: /home/workspace/nuttx/tools/../Make.defs: No such file or directory
> make[2]: *** No rule to make target '/home/workspace/nuttx/tools/../Make.defs'.  Stop.
> make[2]: Leaving directory '/home/workspace/nuttx/tools/pic32'
> make[1]: *** [Makefile.host:276: clean] Error 2
> ```

Thnaks, Fixed on PR#7560 commit 47b2d3df7eed39ad5e2e9646c5c54d86a4f030d4

**Pull request merged**